targets/TARGET_STM/i2c_api.c

```diff
@@ -260,7 +260,12 @@ void i2c_sw_reset(i2c_t *obj)
 
 void i2c_init(i2c_t *obj, PinName sda, PinName scl)
 {
+    memset(obj, 0, sizeof(*obj));
+    i2c_init_internal(obj, sda, scl);
+}
 
+void i2c_init_internal(i2c_t *obj, PinName sda, PinName scl)
```

Comment:
Actually I think you may need to declare this function at the beginning of the file (might be a static one). 
Seems to fail building with a few toolchains